Move Proposals listeners to Wallet Package

Sirmorrison opened the pull request:
* Move proposals initialization from proposals page to state.go
* Move Proposals sync listeners to wallet package.
* Restored the restore button after sync is completed
* Syncs proposal when no proposal has been fetched (initial sync).

ademuanthony commented on fyne/sync.go:
```diff
@@ -47,13 +47,19 @@ func (app *fyneApp) showSyncWindow() {
+			if !syncCompleted {
+				app.loadMainWindowContent()
+				syncCompleted = true
+			}else{
```
This else block can be removed

ademuanthony commented on terminal/pages/sync.go:
```diff
@@ -89,11 +89,18 @@ func LaunchSyncPage(tviewApp *tview.Application, walletMiddleware app.WalletMidd
+			if !syncCompleted {
+				afterSyncing()
+				syncCompleted = true
+			}else{
```
this `else` block can be removed

ademuanthony commented on terminal/pages/sync.go:
```diff
@@ -89,11 +89,18 @@ func LaunchSyncPage(tviewApp *tview.Application, walletMiddleware app.WalletMidd
 }
 
 func startSync(walletMiddleware app.WalletMiddleware, updateStatus func([]string), handleError func(string), afterSyncing func()) {
+	var syncCompleted bool
```
I guess this variable is tracking whether the main page has been loaded or not. Please rename

ademuanthony commented on fyne/sync.go:
```diff
@@ -47,13 +47,19 @@ func (app *fyneApp) showSyncWindow() {
 	app.resizeAndCenterMainWindow(syncWindowContent)
 	app.mainWindow.Show()
 
+	var syncCompleted bool
```
I guess this variable is tracking whether the main page has been loaded or not. Please rename

Sirmorrison replied:
Its tracking is sync was completed. Reason is after sync is completed the process starts again in background.

Sirmorrison committed:
update getProposals() to GetAllProposals()
- Move proposals initialization from proposals page to state.go

- Move proposal listner to wallet package

Sirmorrison committed:
Fix proposals display bug after initial sync
- fix tooltip position on votebar required yes vote percentage indictor
- fix bug with NaN % value when percentage vote is equals to zero

Sirmorrison committed:
Start proposal sync if no proposals are found on app load
* restore refresh button when sync is completed after 3secs from updating text

Sirmorrison committed:
fix proposal state bug
- Proposal state was always on Waiting for admin to start voting,
 fix proposal in discussion state update bug after sync,
 fix bug with yes and no votes not updating after sync

oshorefueled commented on wallet/commands.go:
```diff
@@ -653,17 +653,30 @@ func (wal *Wallet) RenameAccount(walletID int, acct int32, name string, errChan
+	}
+
+}
+
```
This doesn't seem to be used anywhere

oshorefueled commented on ui/proposals_page.go:
```diff
@@ -72,27 +47,25 @@ type tabs struct {
+	wallet           *wallet.Wallet
+	selectedProposal **dcrlibwallet.Proposal
+	proposals        **wallet.Proposals
+	proposalSync     *chan *wallet.NewProposal
```
Is there any reason a pointer to the channel was used here?

Sirmorrison replied:
It is returning the proposal sync status and also the proposal from the proposal listener.

oshorefueled replied:
I don't think you need a pointer to a channel to do that

oshorefueled commented on ui/proposals_page.go:
```diff
@@ -168,81 +142,33 @@ func (pg *proposalsPage) Handle(common pageCommon) {
-func (pg *proposalsPage) onFetchComplete() {
-	if !pg.hasFetchedInitialProposals {
-		pg.hasFetchedInitialProposals = true
+	default:
```
There's no need for a default case if the case is empty

Sirmorrison replied:
Removing the default cases causes the app to freeze when proposals page is selected. I spent the last 30 mins trying to figure out this behavior.

oshorefueled replied:
oh I see. It's a select case and removing the default case causes the app to wait for a signal from the syncedProposal channel.

oshorefueled commented on ui/proposals_page.go:
```diff
@@ -541,24 +496,45 @@ func (pg *proposalsPage) layoutStartSyncSection(gtx C) D {
+			}
+		}
+	}
+}
```
This looks like a duplicate of the code in `addDiscoveredProposal` and the method can simply be called here.

oshorefueled commented on ui/proposals_page.go:
```diff
@@ -308,23 +234,56 @@ func (pg *proposalsPage) layoutTabs(gtx C) D {
 
-func (pg *proposalsPage) layoutErrorState(gtx C) D {
-	str := "Error fetching proposals"
+func (pg *proposalsPage) updateProposal(proposal dcrlibwallet.Proposal) {
```
Kindly add a comment here or a more descriptive method name. What about the proposal is being updated?

oshorefueled commented on wallet/proposal_listener.go:
```diff
@@ -0,0 +1,58 @@
+	VoteFinished
+)
+
+type NewProposal struct {
```
Proposal is just fine here

Sirmorrison committed:
remove unused command.go method
- remove repeating lines of code
- clean up code

Sirmorrison committed:
remove pointer to channel
